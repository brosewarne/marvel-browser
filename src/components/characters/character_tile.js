import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Paper from 'material-ui/Paper';

import { ItemImage } from '../items/item_image';

import './character.css';

/**
 * Render a single CharacterTile
 * @param {Object} character - The character JSON definition
 * @param {String} imgSize - The size of image to use - eg portrait_medium, portrait_uncanny
 * @returns {Node} - The rendered CharacterTile
 * @constructor
 */
export const CharacterTile = ({ character, imgSize = 'portrait_medium' }) => {
    const characterImage = <ItemImage imgSize={imgSize} item={character} />;
    const href = `/character/${character.id}`;
    return (
        <Paper zDepth={4} className="characterTile" key={`${character.id}_tile`}>
            <h1 className="characterTitle">{character.name}</h1>
            <Link
                href={href}
                to={{ pathname: href, state: { character } }}
                title={`Click to see ${character.name}'s profile`}
            >
                {characterImage}
            </Link>
        </Paper>
    );
};

CharacterTile.propTypes = {
    character: PropTypes.object.isRequired,
    imgSize: PropTypes.string.isRequired
};
