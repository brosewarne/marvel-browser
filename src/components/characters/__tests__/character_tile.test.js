import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { CharacterTile } from '../character_tile';

configure({ adapter: new Adapter() });

describe('<CharacterTile />', () => {
    const mockCharacter = {
        id: 1009146,
        name: 'Abomination',
        thumbnail: {
            path: '/a/href/to/an/image',
            extension: 'png'
        }
    };
    it('renders correctly', () => {
        const wrapper = shallow(
            <CharacterTile character={mockCharacter} imgSize="portrait_medium" />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
