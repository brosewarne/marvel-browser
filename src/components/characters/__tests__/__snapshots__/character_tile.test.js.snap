// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterTile /> renders correctly 1`] = `
<WithStyles(Paper)
  className="characterTile"
  key="1009146_tile"
  zDepth={4}
>
  <h1
    className="characterTitle"
  >
    Abomination
  </h1>
  <Link
    href="/character/1009146"
    replace={false}
    title="Click to see Abomination's profile"
    to={
      Object {
        "pathname": "/character/1009146",
        "state": Object {
          "character": Object {
            "id": 1009146,
            "name": "Abomination",
            "thumbnail": Object {
              "extension": "png",
              "path": "/a/href/to/an/image",
            },
          },
        },
      }
    }
  >
    <Component
      imgSize="portrait_medium"
      item={
        Object {
          "id": 1009146,
          "name": "Abomination",
          "thumbnail": Object {
            "extension": "png",
            "path": "/a/href/to/an/image",
          },
        }
      }
    />
  </Link>
</WithStyles(Paper)>
`;
